core/base/marchingTetrahedra/MarchingTetrahedra.h

```diff
@@ -0,0 +1,4047 @@
+     */
+    inline void getCenter(float pos0[3],
+                          float pos1[3],
+                          float pos2[3],
+                          float incenter[3]) const {
+      incenter[0] = 0.3333 * (pos0[0] + pos1[0] + pos2[0]);
+      incenter[1] = 0.3333 * (pos0[1] + pos1[1] + pos2[1]);
+      incenter[2] = 0.3333 * (pos0[2] + pos1[2] + pos2[2]);
+    }
+
+    /**
+     * @brief Get the center of four points
+     *
+     * @param[in] pos0 Position 0
+     * @param[in] pos1 Position 1
+     * @param[in] pos2 Position 2
+     * @param[in] pos3 Position 3
+     * @param[out] incenter Resulting position
+     * @return int 0 on success
+     */
+    inline void getCenter(float pos0[3],
+                          float pos1[3],
+                          float pos2[3],
+                          float pos3[3],
+                          float incenter[3]) const {
+      incenter[0] = 0.25 * (pos0[0] + pos1[0] + pos2[0] + pos3[0]);
+      incenter[1] = 0.25 * (pos0[1] + pos1[1] + pos2[1] + pos3[1]);
+      incenter[2] = 0.25 * (pos0[2] + pos1[2] + pos2[2] + pos3[2]);
+    }
+
+    /**
+     * @brief Interpolate between two points (lambda = 0 -> pos1 / 1 -> pos0)
+     *
+     * @param[in] pos0 Position 0
+     * @param[in] pos1 Position 1
+     * @param[in] lambda Interpolation parameter
+     * @param[out] result Resulting position
+     * @return int 0 on success
+     */
+    inline void interpolatePoints(float pos0[3],
+                                  float pos1[3],
+                                  float lambda,
+                                  float result[3]) const {
+
+      result[0] = lambda * pos0[0] + (1 - lambda) * pos1[0];
+      result[1] = lambda * pos0[1] + (1 - lambda) * pos1[1];
+      result[2] = lambda * pos0[2] + (1 - lambda) * pos1[2];
+    }
+
+  protected:
+    // Output options
+    SURFACE_MODE SurfaceMode{SURFACE_MODE::SM_SEPARATORS};
+
+  private:
+    // Output data
+    SimplexId *output_numberOfPoints_{};
+    std::vector<float> *output_points_{};
+    SimplexId *output_numberOfCells_{};
+    std::vector<unsigned long long> *output_cells_mscIds_{};
+    std::vector<SimplexId> *output_cells_connectivity_{};
+  };
+} // namespace ttk
+
+template <typename dataType, typename triangulationType>
+int ttk::MarchingTetrahedra::execute(const dataType *const scalars,
+                                     const triangulationType &triangulation) {
+
+  Timer t;
+
+  if(scalars == nullptr)
+    return this->printErr("Input scalar field pointer is null.");
+
+  const SimplexId nV = triangulation.getNumberOfVertices();
+  const SimplexId nC = triangulation.getNumberOfCells();
+  const int dim = triangulation.getDimensionality();
+
+  unsigned char *tetCases = nullptr;
+  size_t *numberOfTetCases = nullptr;
+  unsigned long long *cScalars = nullptr;
+
+  tetCases = (unsigned char *)malloc(nC * sizeof(unsigned char));
+  cScalars = (unsigned long long *)malloc(nV * sizeof(unsigned long long));
```

Comment:
Is there a good reason to use `malloc` instead of an smart container like `vector` ?
I assume this related to the free of the memory, this still seems really dangerous. (Where are those `free`d ?)